topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperatorConfig.java

```diff
@@ -117,7 +130,9 @@ public record TopicOperatorConfig(
     static final ConfigParameter<Integer> MAX_BATCH_SIZE = new ConfigParameter<>("STRIMZI_MAX_BATCH_SIZE", strictlyPositive(INTEGER), "100", CONFIG_VALUES);
     static final ConfigParameter<Long> MAX_BATCH_LINGER_MS = new ConfigParameter<>("STRIMZI_MAX_BATCH_LINGER_MS", strictlyPositive(LONG), "100", CONFIG_VALUES);
     static final ConfigParameter<Boolean> ENABLE_ADDITIONAL_METRICS = new ConfigParameter<>("STRIMZI_ENABLE_ADDITIONAL_METRICS", BOOLEAN, "false", CONFIG_VALUES);
-    
+    static final ConfigParameter<String> ALTERABLE_TOPIC_CONFIG = new ConfigParameter<>("STRIMZI_ALTERABLE_TOPIC_CONFIG", STRING, "", CONFIG_VALUES);
```

Comment:
`STRIMZI_RESOURCE_LABELS`, `STRIMZI_TRUSTSTORE_LOCATION`, `STRIMZI_TRUSTSTORE_PASSWORD`, `STRIMZI_KEYSTORE_LOCATION`, `STRIMZI_KEYSTORE_PASSWORD`, `STRIMZI_SASL_MECHANISM`, `STRIMZI_SASL_USERNAME`, `STRIMZI_SASL_PASSWORD`, and `STRIMZI_SECURITY_PROTOCOL` have default values of `""`.

I don't understand how this is different. Are you saying the check would instead of not empty, be not "ALL"?

Reply:
You are right that all the other properties have empty string as default, but with `STRIMZI_ALTERABLE_TOPIC_CONFIG` we are saying: these is the list of topic configurations that can be changed, everything else is forbidden. What if I set this env var to empty string explicitly? Would it deny any configuration change? The answer is no right now. This is why I think it's weird/confusing.

Reply:
I don't recall if ConfigParameter allows null defaults, but if it does perhaps that would be a less ambiguous default.